describe/DAType.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  AbstractDataAttributeDescription,
+  describeAbstractDataAttribute,
+} from "./AbstractDataAttribute.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { ProtNsDescription, describeProtNs } from "./ProtNs.js";
+
+export function isDATypeDescription(
+  type: DATypeDescription
+): type is DATypeDescription {
+  return (
+    (type as DATypeDescription).bdas !== undefined &&
+    (type as DATypeDescription).protns !== undefined
+  );
+}
+
+export interface DATypeDescription extends NamingDescription {
+  /** Optional DAType iedType attribute */
+  iedType?: string;
+  /** `BDA` elements with existing attribute */
+  bdas: Record<string, AbstractDataAttributeDescription>;
+  protns: ProtNsDescription[];
+}
+
+export function DAType(element: Element): DATypeDescription {
+  const daTypeDesc: DATypeDescription = {
+    ...describeNaming(element),
+    bdas: {},
+    protns: [],
+  };
+
+  if (element.getAttribute("iedType"))
+    daTypeDesc.iedType = element.getAttribute("iedType")!;
```

Comment:
Do you want the `iedType` property set if the attribute is set to the empty string?

If so, you could use [hasAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/hasAttribute) as the `if` condition instead.

If not, I'd probably write
```suggestion
  const iedType = element.getAttribute("iedType");
  if (iedType) daTypeDesc.iedType = iedType;
```

Not that `getAttribute` is a significant performance overhead, but it's generally good to be consistent on these things, even just for readability. This pattern shows up in quite a few places, so if you prefer my suggested phrasing, you might want to grep for `getAttribute`.

Reply:
Top! good spotting